packages/react-testing/src/types.ts

```diff
@@ -1,9 +1,13 @@
 import * as React from 'react';
-import {
-  Arguments,
-  MaybeFunctionReturnType,
-  Props as PropsForComponent,
-} from '@shopify/useful-types';
+import {Arguments, MaybeFunctionReturnType} from '@shopify/useful-types';
+
+export type PropsFor<
```

Comment:
this _duplication_ of the old `Props<T>` is a bit ugly, but it's a lot better than using conditional types in each place where we were using `Props<T>` before.